guides/micronaut-k8s-oci/micronaut-k8s-oci.adoc

```diff
@@ -0,0 +1,170 @@
+
+common:jq[]
+
+
+external:micronaut-k8s/create-app-external-guides.adoc[]
+
+== Create Oracle Cloud Kubernetes Cluster
+
+We will use Quick create cluster option on OKE to create Kubernetes Cluster. To start browse https://cloud.oracle.com/containers/clusters/quick[Quick Kubernetes Clusters (OKE)].
+
+image::k8s-oci/create-cluster-wizard.png[]
+
+Choose name for kubernetes cluster, we chose to name it `micronaut-k8s`.
+
+{empty} +
+
+image::k8s-oci/finish-creating-cluster.png[]
+
+Wait to finish all resources to be created.
+
+{empty} +
+
+image::k8s-oci/micronaut-k8s-cluster.png[]
+
+Copy `Cluster Id` you will need it later.
+
+== Prepare and deploy microservices
+
+We will define some variables to make deploying process easier. In `OCI_USER_ID` store your user ocid, you can find it in your oci config file. In `OCI_TENANCY_NAME` store your tenancy name. In `OCI_USERNAME` we will store username in format <tenancy>/<username>. We can reuse `OCI_TENANCY_NAME` and only edit <username> part. Store your region key code in `OCI_REGION` for an example us-phoenix-1. In `OCI_CLUSTER_ID` put the Cluster Id that you copied before.
+
+[source,bash]
+----
+export OCI_USER_ID="ocid1.user.oc1..aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+export OCI_TENANCY_NAME="<tenancy-id>"
+export OCI_USERNAME="$OCI_TENANCY_NAME/<username>"
+export OCI_REGION="<region-key>"
+export OCI_CLUSTER_ID="ocid1.cluster.oc1.iad.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+----
+
+We have to create `AUTH_TOKEN` to be able to authenticate to Oracle Cloud Container Registry.
+
+[source,bash]
+----
+export AUTH_TOKEN=$(oci iam auth-token create --user-id $OCI_USER_ID --description k8s-micronaut | jq -r '.data.token')
+----
+
+Run next command to log in to `ocir.io` (Oracle Cloud Container Registry):
+
+[source,bash]
+----
+docker login $OCI_REGION.ocir.io -u $OCI_USERNAME -p $AUTH_TOKEN
+----
+
+=== Users microservice
+
+external-template:micronaut-k8s-oci/k8s-microservice.adoc[arg0=users]
+
+=== Orders microservice
+
+external-template:micronaut-k8s-oci/k8s-microservice.adoc[arg0=orders]
+
+=== API microservice
+
+external-template:micronaut-k8s-oci/k8s-microservice.adoc[arg0=api]
+
+=== Deploy services to OKE
+
+Create folder for `kubectl` configuration.
+
+[source,bash]
+----
+mkdir -p $HOME/.kube
+----
+
+Generate `kubectl` configuration for authentication to OKE.
+
+[source,bash]
+----
+oci ce cluster create-kubeconfig --cluster-id $OCI_CLUSTER_ID --file $HOME/.kube/config --region $OCI_REGION --token-version 2.0.0  --kube-endpoint PUBLIC_ENDPOINT
+----
+
+Set `KUBECONFIG` to created config file. This variable is consumed by `kubectl`.
+
+[source,bash]
+----
+export KUBECONFIG=$HOME/.kube/config
+----
+
+Create `ocirsecret` secret that will be used for authentication to Oracle Cloud Container Registry. This is needed because OKE needs docker credentials to be able to pull microservices images.
```

Comment:
```suggestion
Create `ocirsecret` secret that will be used for authentication to Oracle Cloud Container Registry. This is needed because OKE needs credentials to be able to pull microservices images.
```